### PR TITLE
Added tests and a major refactor

### PR DESCRIPTION
**Pull request by NuclearNic**
* Tests use a mock-app (called an engine here) for the tests. Perhaps they are still a bit lacking? Didn't find these the easiest to write

* Refactor: Use a ZipkinConfig class instead of calling ENVs all over the place

### CODE REVIEW

File: lib/rails_opentracer.rb [hunk 1]
```diff
@@ -7,32 +8,23 @@
...
-    def instrument(tracer: OpenTracing.global_tracer, active_span: nil, active_record: true)
-      ActiveRecord::RailsOpentracer.instrument(tracer: tracer, active_span: active_span) if active_record
-    end
-
-    def disable
-      ActiveRecord::RailsOpentracer.disable
-    end
-  end
-
+
```
Comment (emile-jumo) [inline lib/rails_opentracer.rb]: [nit] remove empty line

File: lib/rails_opentracer/middleware.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 require 'zipkin'
+require 'rails_opentracer/zipkin_config'
 
 module RailsOpentracer
-
   class Railtie < Rails::Railtie
+
```
Comment (emile-jumo) [inline lib/rails_opentracer/middleware.rb]: [nit] remove empty line

File: lib/rails_opentracer/zipkin_config.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+module RailsOpentracer
+  class ZipkinConfig
+
```
Comment (emile-jumo) [inline lib/rails_opentracer/zipkin_config.rb]: [nit] remove empty line

File: lib/rails_opentracer/zipkin_config.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    def self.zipkin_url
+      if Rails.env == "test" || Rails.env == "development"
+        'http://localhost:9411'
+      else
+        ENV['ZIPKIN_SERVICE_URL']
+      end
+    end 
+  end
+end
```
Comment (emile-jumo) [inline lib/rails_opentracer/zipkin_config.rb]: [nit] add eof new line

File: test/internal/log/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+*.log
```
Comment (emile-jumo) [inline test/internal/log/.gitignore]: [nit] add eof new line

File: test/rails_opentracer_combustion_test.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
+require 'test_helper'
+require "combustion"
+Combustion.path = "test/internal"
+Combustion.initialize! :all
+
+# TODO: WHEN ENV is test then don't ping zipkin client url
+
+class RailsOpentracerCombustionTest < Minitest::Test
+
```
Comment (emile-jumo) [inline test/rails_opentracer_combustion_test.rb]: [nit] remove empty line

File: test/rails_opentracer_combustion_test.rb [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    test_controller = Combustion::Application::PagesController.new
+    span = test_controller.index
+    refute_nil span
+  end
+
+  def test_active_record_query_creates_span
+    args = ["","","","", {:sql=>"", :name=>"", :connection_id=>12345} ]
+    refute_nil ActiveRecord::RailsOpentracer.sql(args: args)
+  end
+end
```
Comment (emile-jumo) [inline test/rails_opentracer_combustion_test.rb]: [nit] add eof new line

File: test/rails_opentracer_combustion_test.rb [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  def test_get_request_has_span
+    test_controller = Combustion::Application::PagesController.new
+    span = test_controller.index
+    refute_nil span
+  end
+
+  def test_active_record_query_creates_span
+    args = ["","","","", {:sql=>"", :name=>"", :connection_id=>12345} ]
```
Comment (emile-jumo) [inline test/rails_opentracer_combustion_test.rb]: Rather changes to:
```
{sql: "", name: "", connection_id: 12345}
```

File: test/rails_opentracer_test.rb [hunk 9]
```diff
@@ -1,11 +1,9 @@
-require "test_helper"
+require 'test_helper'
+require 'minitest/spec'
 
 class RailsOpentracerTest < Minitest::Test
+
```
Comment (emile-jumo) [inline test/rails_opentracer_test.rb]: [nit] remove empty line

File: test/rails_opentracer_combustion_test.rb (hunk 8 above)
Comment (emile-jumo) [reply test/rails_opentracer_combustion_test.rb]: and add space after `,`

File: lib/rails_opentracer/zipkin_config.rb [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    def self.opentracer_enabled?
+     ENV['RAILS_OPENTRACER_ENABLED'] == 'yes' 
+    end
+
+    def self.opentracer_enabled_and_zipkin_url_present?
+      ENV['RAILS_OPENTRACER_ENABLED'] == 'yes' && ENV['ZIPKIN_SERVICE_URL'].present?
+    end
+
+    def self.zipkin_url
+      if Rails.env == "test" || Rails.env == "development"
```
Comment (emile-jumo) [inline lib/rails_opentracer/zipkin_config.rb]: [Suggestion] Maybe chance to 
```
if Rails.env.test? || Rails.env.development?
```

Comment (NuclearNic) [reply lib/rails_opentracer/zipkin_config.rb]: haha yeah wow that's bad of me

**Pull request merged**